### PR TITLE
Realized own JUnit tests

### PR DESCRIPTION
**Pull request by gigbat**

### CODE REVIEW

File: src/main/java/core/basesyntax/Calculator.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
+package core.basesyntax;
+
+public class Calculator {
```
Comment (Ketch92) [inline src/main/java/core/basesyntax/Calculator.java:3]: I see your realization and found it a bit complicated as for a single method. It's totally up to you to keep it or simplify.

File: src/main/java/core/basesyntax/Calculator.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    public boolean checkForNull() {
+        if (firstNumber == null || secondNumber == null) {
+            throw new NullPointerException();
+        }
+        return false;
+    }
+
+    public Integer addition(Integer a, Integer b) {
+        result = a + b;
+        return result;
```
Comment (Ketch92) [inline src/main/java/core/basesyntax/Calculator.java:22]:
```suggestion
        return a + b;
```

Comment (Ketch92) [reply src/main/java/core/basesyntax/Calculator.java:22]: and for the rest of the methods, the same comment is appicable

File: src/main/java/core/basesyntax/Calculator.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        if (a < b && a != 0) {
+            result = b / a;
+            return result;
+        }
+        if (a > b && b != 0) {
+            result = a / b;
+            return result;
+        }
+        return 1;
+    }
```
Comment (Ketch92) [inline src/main/java/core/basesyntax/Calculator.java:40]: I would say it's weird result when you swap the divisor and divider, if I put a / b I want that to have it, but no way I expect b/a

Comment (Ketch92) [reply src/main/java/core/basesyntax/Calculator.java:40]: Also here you can throw an Arithmetic exception when input is a/0, but it's up to you

File: src/main/java/core/basesyntax/Calculator.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+public class Calculator {
+    private Integer result;
+    private Integer firstNumber;
+    private Integer secondNumber;
+
+    public Calculator(Integer firstNumber, Integer secondNumber) {
+        this.firstNumber = firstNumber;
+        this.secondNumber = secondNumber;
+    }
```
Comment (Ketch92) [inline src/main/java/core/basesyntax/Calculator.java:11]: Do you think every calculator should have preset values? I mean, like in real... would you like it if you buy a calculator that works with 5 and 10 values?

File: src/main/java/core/basesyntax/Calculator.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+    public boolean checkForNull() {
+        if (firstNumber == null || secondNumber == null) {
+            throw new NullPointerException();
+        }
+        return false;
```
Comment (Ketch92) [inline src/main/java/core/basesyntax/Calculator.java:17]: will ever this method return true?

File: src/main/java/core/basesyntax/Calculator.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            result = a / b;
+            return result;
+        }
+        return 1;
+    }
+
+    public Integer multiplication(Integer a, Integer b) {
+        result = a * b;
+        return result;
+    }
```
Comment (Ketch92) [inline src/main/java/core/basesyntax/Calculator.java:45]: in general, if you accepted that kind of solution it would look better if you can work with it like builer
new Calculator(5, 1000).multiplication() and for multiplication, you already set values in constructor, why should I pass them twice. Then result parameter is redundant in any form I think

File: src/main/java/core/basesyntax/Calculator.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        return firstNumber;
+    }
+
+    public boolean setFirstNumber(Integer firstNumber) {
+        if (firstNumber == null) {
+            throw new NullPointerException();
+        }
+        this.firstNumber = firstNumber;
+        return true;
+    }
```
Comment (Ketch92) [inline src/main/java/core/basesyntax/Calculator.java:57]: but in constructor, you can pass null freely...

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    @Test
+    void isSet_SecondValue() {
+        Assertions.assertTrue(calculate.setSecondNumber(10));
+    }
+
+    @Test
+    void checkForConsistent_firstValue() {
+        Integer expected = calculate.getFirstNumber();
+        Assertions.assertEquals(expected, calculate.getFirstNumber(), "You number not equals");
+    }
```
Comment (Ketch92) [inline src/test/java/core/basesyntax/CalculatorTest.java:51]: Do you see a mistake here? Sorry, I can't say what is the mistake, but think about it please, and for other similar cases the same

**Review by Ketch92 — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 9]
```diff
@@ -0,0 +1,102 @@
+package core.basesyntax;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class CalculatorTest {
```
Comment (Ketch92) [inline src/test/java/core/basesyntax/CalculatorTest.java:7]: your tests should also check for critical values ranges, what will happen if you put there maximal value of int?

**Review by Ketch92 — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Calculator.java (hunk 2 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/Calculator.java:22]: done

File: src/main/java/core/basesyntax/Calculator.java (hunk 3 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/Calculator.java:40]: done

File: src/main/java/core/basesyntax/Calculator.java (hunk 4 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/Calculator.java:11]: done

File: src/main/java/core/basesyntax/Calculator.java (hunk 5 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/Calculator.java:17]: done

File: src/main/java/core/basesyntax/Calculator.java (hunk 6 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/Calculator.java:45]: I have just removed the previous constructor and make the constructor without arguments. Customer will input values using the method "setters" set into class Calculator

File: src/main/java/core/basesyntax/Calculator.java (hunk 7 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/Calculator.java:57]: Made constructor non-arguments

File: src/main/java/core/basesyntax/Calculator.java (hunk 1 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/Calculator.java:3]: :)))

File: src/test/java/core/basesyntax/CalculatorTest.java (hunk 9 above)
Comment (gigbat) [reply src/test/java/core/basesyntax/CalculatorTest.java:7]: done

**Review by Ketch92 — APPROVED** [review]

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]
image

File: src/main/java/core/basesyntax/Calculator.java [hunk 10]
```diff
@@ -0,0 +1,66 @@
+package core.basesyntax;
+
+public class Calculator {
+    private Double firstNumber;
+    private Double secondNumber;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Calculator.java:5]: why do you need this state? let's just have one method calculate and do all checks there(you still can use private methods)

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+    void multiplication_Final() {
+        calculator.setFirstNumber(MAX_VALUE);
+        calculator.setSecondNumber(MIN_VALUE);
+        double expected = MAX_VALUE * MIN_VALUE;
+        double actual = calculator.calculate(MAX_VALUE, MIN_VALUE, '*');
+        Assertions.assertEquals(expected, actual);
+    }
+
+    @Test
+    void power_withoutFinal() {
```
Comment (dmytro-hryhoruk) [inline src/test/java/core/basesyntax/CalculatorTest.java:135]: what does mean withoutFinal/final in tests naming?

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+
+        temp = new Calculator(12.0, null);
+        calculator = temp;
+        Assertions.assertThrows(NullPointerException.class, () -> calculator.checkForNull());
+    }
+
+    @Test
+    void addition_withoutFinal() {
+        double firstNumber = Math.random() * MAX_VALUE;
+        double secondNumber = Math.random() * MIN_VALUE;
```
Comment (dmytro-hryhoruk) [inline src/test/java/core/basesyntax/CalculatorTest.java:49]: let's not use random tests. It always produces a different test case and sometimes your test may pass and some it's not. Even though the code that is tested has not been changed.

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]
The method naming isn't obvious.
Remove the state from the Calculator
let's add these test cases:
- raising positive/negative value to positive power
- raising positive/negative value to negative power
- raising positive/negative value to zero power
- raising zero to power

File: src/main/java/core/basesyntax/Calculator.java (hunk 10 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/Calculator.java:5]: done

File: src/test/java/core/basesyntax/CalculatorTest.java (hunk 11 above)
Comment (gigbat) [reply src/test/java/core/basesyntax/CalculatorTest.java:135]: renamed

File: src/test/java/core/basesyntax/CalculatorTest.java (hunk 12 above)
Comment (gigbat) [reply src/test/java/core/basesyntax/CalculatorTest.java:49]: okey, done

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]
Let's add test cases for negative numbers(don't forget to raise to negative power)

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+    }
+
+    @Test
+    void power_negative() {
+        double firstNumber = -2.0;
+        double secondNumber = -3.0;
+        double expected = Math.pow(firstNumber, secondNumber);
+        double actual = calculator.calculate(firstNumber, secondNumber, '^');
+        Assertions.assertEquals(expected, actual);
+    }
```
Comment (dmytro-hryhoruk) [inline src/test/java/core/basesyntax/CalculatorTest.java:68]: lets add a few test cases with 0^(-2), -2^0.5 etc.
Also let's group our tests by operation: test on power operation near each other, same with other operations

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/CalculatorTest.java (hunk 13 above)
Comment (gigbat) [reply src/test/java/core/basesyntax/CalculatorTest.java]: done

**Review by dmytro-hryhoruk — APPROVED** [review]
